hits_ingest.py

```diff
@@ -64,25 +109,56 @@
-
-# follow a similar process to ingest calibration products if the doIngestCalibs flag is set to True
-# WORK IN PROGRESS! TODO:
-# - get the command-line task to work on its own
-# - get this script to work in the same way
-# - find a reasonable way to ingest all the calibration products (bias, flat, defect, etc.)
-if doIngestCalibs == True:
-    print('Ingesting calibration products...')
-    args = [repodir, '--calib', calibrepodir, '--calibType', 'defect', '--validity', '999']
+
+    print('Images are now ingested in {0}'.format(repo))
+
+# follow a similar process to ingest calibrations for doIngestCalibs = True
+# catch the common sqlite3.IntegrityError and print some useful information
+# NOTE: I chose to ingest the 'defect' calibration products separately on
+#       the command line by specifying a different directory and adding the
+#       argument --calibType defect. In the future, this should be done
+#       here alongside ingesting the biases and flats.
+if doIngestCalibs:
+    print('Ingesting calibration products...')  # just biases and flats for now
+    args = [repo, '--calib', calibrepo, '--validity', '999']
     args.extend(datafiles)
-    argumentParser = ingest.DecamIngestArgumentParser(name='ingestCalibs')
-    config = IngestConfig()
+    argumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
+    config = IngestCalibsConfig()
     config.parse.retarget(ingestCalibs.DecamCalibsParseTask)
-    #ingestTask = ingest.DecamIngestTask(config=config)
-    ingestTask = ingestCalibs.DecamCalibsParseTask(config=config, name='ingestCalibs')
+    ingestTask = IngestCalibsTask(config=config, name='ingestCalibs')
     parsedCmd = argumentParser.parse_args(config=config, args=args)
-    ## ^^ this fails because lsst.pipe.tasks.ingest.RegisterConfig has no attribute detector(??)
-    ingestTask.run(parsedCmd)
-    print('Calibrations from {0} for {1} are now ingested in {2}'.format(datadir, repodir, calibrepodir))
+    try:
+        ingestTask.run(parsedCmd)
+    except sqlite3.IntegrityError as detail:
+        print('~~~ !!! ~~~')
+        print('sqlite3.IntegrityError: ', detail)
+        print('(sqlite3 doesn\'t think all the calibration files are unique)')
+        print('If this isn\'t your first time ingesting these calibration')
+        print('  files, move or delete the existing database and try again.')
+        print('If this is your first time ingesting these calibration files,')
+        print('  make sure you only use image or wtmap MasterCals, not both.')
+        print('~~~ !!! ~~~')
+    else:
+        print('Success!')
+        print('Calibrations corresponding to {0} are now ingested in {1}' \
+                                                    .format(repo, calibrepo))
+
+# the final step is to run processCcd to do ISR by combining the ingested
+# images and calibration products
+if doProcessCcd:
+    print('Running ProcessCcd...')
+    OBS_DECAM_DIR = os.getenv('OBS_DECAM_DIR')
+    os.chdir(calibrepo)
```

Comment:
As far as I can tell, this is a consequence of DM-5466, which I really do plan to tackle soon. I don't like having the chdir in here either so I will add a comment/note to revisit this once obs_decam is a little smarter about calib paths.